B006 (mutable-argument-default) no longer offers autofix even with --unsafe-fixes

<!--
Thank you for taking the time to report an issue! We're glad to have you involved with Ruff.

If you're filing a bug report, please consider including the following information:

* A minimal code snippet that reproduces the bug.
* The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.
* The current Ruff settings (any relevant sections from your `pyproject.toml`).
* The current Ruff version (`ruff --version`).
-->

Hello 👋 

B006 (`mutable-argument-default`) no longer offers an autofix even when `--unsafe-fixes` is passed.

Use the example from `ruff rule B006` to reproduce this:

```python
# /tmp/bad.py
def add_to_list(item, some_list=[]):
    some_list.append(item)
    return some_list
```

At HEAD (df807ff91248c896cf7f11b25e4f64ecb1d77066) and v0.1.1:

```sh
❯ cargo run -p ruff_cli -- check --select B --unsafe-fixes /tmp/bad.py --no-cache
/tmp/bad.py:1:33: B006 Do not use mutable data structures for argument defaults
Found 1 error.
```

I've bisected this to 22e18741bdf58bc15a69da2503d7b48443e5a038, but that may only be where the issue was surfaced in the CLI.

The output is slightly different in that commit, there's an additional line:

```sh
/tmp/bad.py:1:33: B006 Do not use mutable data structures for argument defaults
Found 1 error.
[*] 0 fixable with the --fix option.
```

If I omit `--unsafe-fixes` there's a hint about a hidden fix:

```sh
❯ cargo run -p ruff_cli -- check --select B /tmp/bad2.py --no-cache
/tmp/bad.py:1:33: B006 Do not use mutable data structures for argument defaults
Found 1 error.
1 hidden fix can be enabled with the `--unsafe-fixes` option.
```

(I don't see this hint in 0.1.1 or HEAD, but I haven't bisected where it went away.)

In the parent of the bisected commit and in v0.0.292, I get the expected output:

```sh
/tmp/bad.py:1:33: B006 [*] Do not use mutable data structures for argument defaults
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

(I noticed after the fact that I ran Ruff from its repository which has Ruff config in pyproject.toml, so that could have skewed results a little. But the repro should still be valid.)

--- a/crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs
+++ b/crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs
@@ -30,6 +30,12 @@ use crate::checkers::ast::Checker;
 /// Types outside of the standard library can be marked as immutable with the
 /// [`flake8-bugbear.extend-immutable-calls`] configuration option.
 ///
+/// ## Known problems
+/// Mutable argument defaults can be used intentionally to cache computation
+/// results. Replacing the default with `None` or an immutable data structure
+/// does not work for such usages. Instead, prefer the `@functools.lru_cache`
+/// decorator from the standard library.
+///
 /// ## Example
 /// ```python
 /// def add_to_list(item, some_list=[]):

--- a/crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs
+++ b/crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs
@@ -197,5 +197,5 @@ fn move_initialization(
     }
 
     let initialization_edit = Edit::insertion(content, pos);
-    Some(Fix::display_edits(default_edit, [initialization_edit]))
+    Some(Fix::unsafe_edits(default_edit, [initialization_edit]))
 }

